hydra-node/src/Hydra/Chain/Direct/GenesisParameters.hs

```diff
@@ -0,0 +1,75 @@
+import Hydra.Cardano.Api
+import Hydra.Cardano.Api.Prelude (toShelleyLovelace, toShelleyNetwork)
+
+newtype GenesisTranslationError = GenesisTranslationError String deriving (Eq, Show)
+
+instance Exception GenesisTranslationError
+
+toShelleyGenesis :: GenesisParameters -> IO (ShelleyGenesis LedgerEra)
+toShelleyGenesis genesis =
+  case toShelleyGenesisMaybe genesis of
+    Nothing -> throwIO $ GenesisTranslationError "Could not translate GenesisParameters to ShelleyGenesis ShelleyEra"
+    (Just shelleyGenesis) -> pure shelleyGenesis
+
+toShelleyGenesisMaybe :: GenesisParameters -> Maybe (ShelleyGenesis LedgerEra)
+toShelleyGenesisMaybe GenesisParameters{..} = do
+  slotsCoeff <- boundRational protocolParamActiveSlotsCoefficient
+  shelleyPParams <- toShelleyPParams protocolInitialUpdateableProtocolParameters
+  pure
+    ShelleyGenesis
+      { sgSystemStart = protocolParamSystemStart
+      , sgNetworkMagic = unNetworkMagic . toNetworkMagic $ protocolParamNetworkId
+      , sgNetworkId = toShelleyNetwork protocolParamNetworkId
+      , sgActiveSlotsCoeff = slotsCoeff
+      , sgSecurityParam = fromIntegral protocolParamSecurity
+      , sgEpochLength = protocolParamEpochLength
+      , sgSlotsPerKESPeriod = fromIntegral protocolParamSlotsPerKESPeriod
+      , sgMaxKESEvolutions = fromIntegral protocolParamMaxKESEvolutions
+      , sgSlotLength = protocolParamSlotLength
+      , sgUpdateQuorum = fromIntegral protocolParamUpdateQuorum
+      , sgMaxLovelaceSupply = fromIntegral protocolParamMaxLovelaceSupply
+      , sgProtocolParams = shelleyPParams
+      , sgGenDelegs = mempty
+      , sgInitialFunds = mempty
+      , sgStaking = emptyGenesisStaking
+      }
+
+toShelleyPParams :: ProtocolParameters -> Maybe (ShelleyPParams LedgerEra)
+toShelleyPParams ProtocolParameters{..} = do
+  pledgeInfluence <- boundRational protocolParamPoolPledgeInfluence
+  monetaryExpansion <- boundRational protocolParamMonetaryExpansion
+  treasuryCut <- boundRational protocolParamTreasuryCut
+  decentralization <- boundRational =<< protocolParamDecentralization
+  pure $
+    ShelleyPParams
+      { _minfeeA = protocolParamTxFeePerByte
+      , _minfeeB = protocolParamTxFeeFixed
+      , _maxBBSize = protocolParamMaxBlockBodySize
+      , _maxTxSize = protocolParamMaxTxSize
+      , _maxBHSize = protocolParamMaxBlockHeaderSize
+      , _keyDeposit = toShelleyLovelace protocolParamStakeAddressDeposit
+      , _poolDeposit = toShelleyLovelace protocolParamStakePoolDeposit
+      , _eMax = protocolParamPoolRetireMaxEpoch
+      , _nOpt = protocolParamStakePoolTargetNum
+      , _a0 = pledgeInfluence
+      , _rho = monetaryExpansion
+      , _tau = treasuryCut
+      , _d = decentralization
+      , _extraEntropy = Ledger.NeutralNonce
+      , _protocolVersion =
+          let (a, b) = protocolParamProtocolVersion
+           in Ledger.ProtVer a b
+      , _minUTxOValue =
+          maybe (error "missing minUTxO value") toShelleyLovelace protocolParamMinUTxOValue
```

Comment:
This is partial. So if we do have an `error` here, we could also have errors for all the bounded rationals.